src/Authentication/Authentication/custom/common/Permissions.ps1

```diff
@@ -34,7 +34,20 @@ function Permissions_GetPermissionsData([bool] $online) {
     # Make a REST request to MS Graph to get the permissions data from the Microsoft Graph service principal
     if ( $online -or ! $_permissions.msGraphServicePrincipal -or ! $_permissions.isFromInvokeMgGraphRequest ) {
         try {
-            $restResult = Invoke-MgGraphRequest -method GET -OutputType PSObject $_permissions.msGraphPermissionsRequestUri
+            
+            # Get-MgContext is used to get the current context for the request to MS Graph
+            # From the context, we can get the current environment and use it to get the permissions request URI
+            # If the context is not available, then we will use the default permissions request URI
+
+            $context = Get-MgContext
+            $uri = $_permissions.msGraphPermissionsRequestUri
+            if($context){
+                $currentEnv = $context.Environment
+                $allEnv = Get-MgEnvironment
+                $env = $allEnv | Where-Object { $_.Name -eq $currentEnv }
+                $uri = $env.GraphEndpoint + "/v1.0/servicePrincipals?`$filter=appId eq '$Permissions_msGraphApplicationId'"
+            }
```

Comment:
Not sure but would it be better if this happens under `_Permissions_Initialize`? Esp in case the other methods also need to call Graph

Reply:
When you import authentication module into the session i.e. ``Import-Module Microsoft.Graph.Authentication``, you are loading ``Find-MgGraphPermissions`` cmdlet which [loads](https://github.com/microsoftgraph/msgraph-sdk-powershell/blob/fddf88ca79b9dcefb51cee1f9a4d22e0df5e94d6/src/Authentication/Authentication/custom/Find-MgGraphPermission.ps1#L7) the ``Permissions.ps1`` script. Therefore, having that block in that function will result to null stack traces thrown in a PowerShell session because at that point a connection to graph has not yet been done. i.e. Null graph context in the session. Keep in mind there are some users who will have the authentication import step and may have enforced strict error handling in their scripts. In short that code block only comes into play when a connection to graph has been established and the context set in the session.